src/Microsoft.DotNet.Kusto/KustoClientProvider.cs

```diff
@@ -54,16 +53,44 @@ public ICslQueryProvider GetProvider()
             if (value != null)
                 return value;
 
-            _kustoQueryProvider = value = KustoClientFactory.CreateCslQueryProvider(_options.CurrentValue.QueryConnectionString);
+            _kustoQueryProvider = value = KustoClientFactory.CreateCslQueryProvider(GetKustoConnectionStringBuilder());
             return value;
         }
     }
 
     private string DatabaseName => _options.CurrentValue.Database;
+    private string KustoClusterUri => _options.CurrentValue.KustoClusterUri;
+    private string ManagedIdentityId => _options.CurrentValue.ManagedIdentityId;
+    private bool UseAzCliAuthentication => _options.CurrentValue.UseAzCliAuthentication;
+
+    private KustoConnectionStringBuilder GetKustoConnectionStringBuilder()
```

Comment:
Suggest making this a `public static` helper e.g., `GetKustoConnectionStringBuilder(OptionsMonitor<KustoOptions> options)` to avoid duplication